src/test/kotlin/com/parmet/buf/gradle/AbstractGenerateTest.kt

```diff
@@ -38,4 +38,21 @@ abstract class AbstractGenerateTest : AbstractBufIntegrationTest() {
             listOf("build", "bufbuild", "generated", "java", "com", "google", "type", "DateTime.java")
         assertThat(Paths.get(projectDir.absolutePath, *generatedPathElements.toTypedArray()).toFile().exists()).isTrue()
     }
+
+    @Test
+    fun `generate java with gen file override`() {
```

Comment:
Small thing: I think this test should be named `buf generate with buf gen template file override`.